pkg/k8s/annotate.go

```diff
@@ -42,43 +43,44 @@ type K8sCiliumClient struct {
 	clientset.Interface
 }
 
-func updateNodeAnnotation(c kubernetes.Interface, node *v1.Node, v4CIDR, v6CIDR *cidr.CIDR, v4HealthIP, v6HealthIP, v4CiliumHostIP, v6CiliumHostIP net.IP) (*v1.Node, error) {
-	if node.Annotations == nil {
-		node.Annotations = map[string]string{}
-	}
+func updateNodeAnnotation(c kubernetes.Interface, nodeName string, v4CIDR, v6CIDR *cidr.CIDR, v4HealthIP, v6HealthIP, v4CiliumHostIP, v6CiliumHostIP net.IP) error {
+	annotations := map[string]string{}
 
 	if v4CIDR != nil {
-		node.Annotations[annotation.V4CIDRName] = v4CIDR.String()
+		annotations[annotation.V4CIDRName] = v4CIDR.String()
 	}
 	if v6CIDR != nil {
-		node.Annotations[annotation.V6CIDRName] = v6CIDR.String()
+		annotations[annotation.V6CIDRName] = v6CIDR.String()
 	}
 
 	if v4HealthIP != nil {
-		node.Annotations[annotation.V4HealthName] = v4HealthIP.String()
+		annotations[annotation.V4HealthName] = v4HealthIP.String()
 	}
 	if v6HealthIP != nil {
-		node.Annotations[annotation.V6HealthName] = v6HealthIP.String()
+		annotations[annotation.V6HealthName] = v6HealthIP.String()
 	}
 
 	if v4CiliumHostIP != nil {
-		node.Annotations[annotation.CiliumHostIP] = v4CiliumHostIP.String()
+		annotations[annotation.CiliumHostIP] = v4CiliumHostIP.String()
 	}
 
 	if v6CiliumHostIP != nil {
-		node.Annotations[annotation.CiliumHostIPv6] = v6CiliumHostIP.String()
+		annotations[annotation.CiliumHostIPv6] = v6CiliumHostIP.String()
 	}
 
-	node, err := c.CoreV1().Nodes().Update(node)
-	if err != nil {
-		return nil, err
+	if len(annotations) == 0 {
+		return nil
 	}
 
-	if node == nil {
-		return nil, ErrNilNode
+	raw, err := json.Marshal(annotations)
+	if err != nil {
+		return err
 	}
+	patch := []byte(fmt.Sprintf(`{"metadata":{"annotations":%s}}`, raw))
 
-	return node, nil
+	_, err = c.CoreV1().Nodes().Patch(nodeName, types.StrategicMergePatchType, patch)
```

Comment:
Why don't we need to check
```
type ServerCapabilities struct {
        // Patch is the ability to use PATCH to modify a resource
        Patch bool
```
?

Reply:
Patch issues were only problematic for custom resources not k8s data structures. I test it with k8s 1.10 and worked fine.